### PR TITLE
checkTx endpoint changed to checkTxs

### PR DESCRIPTION
**Pull request by thomasgonda3**

### CODE REVIEW

File: src/indexApi.ts [hunk 1]
```diff
@@ -54,6 +58,14 @@ const asAppIdDbReq = asObject({
   appId: asString
 })
 
+interface CheckTxs {
```
Comment (thomasgonda3) [inline src/indexApi.ts:61]: interface checkTxsResponse

Comment (thomasgonda3) [reply src/indexApi.ts:61]: Check

File: src/indexApi.ts [hunk 2]
```diff
@@ -16,10 +16,14 @@ const asAnalyticsReq = asObject({
...
 })
 
-const asCheckTxReq = asObject({
-  apiKey: asString,
-  pluginId: asString,
-  orderId: asString
+const asCheckTxsReq = asObject({
+  data: asArray(
+    asObject({
+      apiKey: asString,
```
Comment (thomasgonda3) [inline src/indexApi.ts:22]: move apiKey outside of each individual tx, and check it only once.

File: src/indexApi.ts [hunk 3]
```diff
@@ -54,6 +58,14 @@ const asAppIdDbReq = asObject({
   appId: asString
 })
 
+interface CheckTxs {
+  appId?: string
```
Comment (thomasgonda3) [inline src/indexApi.ts:62]: appId is returned outside of array of responses.

File: src/indexApi.ts [hunk 4]
```diff
@@ -150,45 +162,56 @@ async function main(): Promise<void> {
...
+        appId: undefined,
+        pluginId: tx.pluginId,
+        orderId: tx.orderId,
+        error: undefined,
+        usdValue: undefined
+      }
+      const searchedAppId = apps.find(app => app._id === tx.apiKey)
+      if (typeof searchedAppId === 'undefined') {
+        returnTx.error = 'Api Key has no match.'
+        data.push(returnTx)
```
Comment (thomasgonda3) [inline src/indexApi.ts:185]: apiKey if wrong should throw whole thing

File: src/indexApi.ts [hunk 5]
```diff
@@ -150,45 +162,56 @@ async function main(): Promise<void> {
...
+        continue
+      }
+      const appId = searchedAppId.appId
+      returnTx.appId = appId
+      let result
+      try {
+        const query = `${appId}_${tx.pluginId}:${tx.orderId}`.toLowerCase()
+        const dbResult = await reportsTransactions.get(query)
```
Comment (thomasgonda3) [inline src/indexApi.ts:193]: should be bulk get

Comment (thomasgonda3) [reply src/indexApi.ts:193]: in batches of 100

File: src/indexApi.ts [hunk 6]
```diff
@@ -16,12 +23,24 @@ const asAnalyticsReq = asObject({
...
-  orderId: asString
+  data: asArray(
+    asObject({
+      pluginId: asString,
+      orderId: asString
+    })
+  )
 })
 
+const asCheckTxsServerResponse = asArray(
```
Comment (paullinator) [inline src/indexApi.ts:36]: This really should be `asTxFetch`. This is not a server response but rather the response of the database.

File: src/indexApi.ts [hunk 7]
```diff
@@ -150,45 +178,54 @@ async function main(): Promise<void> {
...
-      } else {
-        res.status(500).send(`Internal Server Error.`)
-        return
+    const { appId } = searchedAppId
+    const data: CheckTxsResponse[] = []
+    let counter = 0
+    while (counter < queryResult.data.length) {
+      const currentBatch = queryResult.data.slice(
+        counter,
+        counter + CHECKTXS_BATCH_LIMIT
```
Comment (paullinator) [inline src/indexApi.ts:198]: The batching is supposed to be enforced at the API level meaning the utility server should send no more than 100 order IDs to the report server. Within the checkTxs endpoint, you do not need to batch queries to the database. However you should throw a 400 error if the caller sends more than 100 order IDs.

**Review by paullinator — CHANGES_REQUESTED** [review]
See comments. As well, please squash all commits into one for this pr.

**Review by paullinator — APPROVED** [review]

**Pull request merged**